packages/sdk/test/common/anvil.ts

```diff
@@ -0,0 +1,283 @@
+  return {
+    port,
+    rpcUrls: {
+      default: {
+        // append the test worker id to the rpc urls.
+        http: [`http://localhost:${port}/${pool}`],
+        webSocket: [`ws://localhost:${port}/${pool}`],
+      },
+      public: {
+        http: [`http://localhost:${port}/${pool}`],
+        webSocket: [`ws://localhost:${port}/${pool}`],
+      },
+    },
+  } as const;
+}
+
+type Fork = { blockNumber: bigint; url: string; opStack?: boolean };
+
+export type AnvilChain = Compute<
+  Chain & {
+    fork: Fork;
+    port: number;
+  }
+>;
+
+const optimismFork = {
+  blockNumber: BLOCK_NUMBER_OPTIMISM,
+  url: FORK_URL_OPTIMISM,
+  opStack: true,
+} as const satisfies Fork;
+
+const baseFork = {
+  blockNumber: BLOCK_NUMBER_BASE,
+  url: FORK_URL_BASE,
+  opStack: true,
+} as const satisfies Fork;
+
+const mainnetFork = {
+  blockNumber: BLOCK_NUMBER_MAINNET,
+  url: FORK_URL_MAINNET,
+} as const satisfies Fork;
+
+const arbitrumFork = {
+  blockNumber: BLOCK_NUMBER_ARBITRUM,
+  url: FORK_URL_ARBITRUM,
+} as const satisfies Fork;
+
+//  PORT: 8545
+const optimismAnvil = {
+  ...optimism,
+  ...getRpcUrls({ port: 8545 }),
+  fork: optimismFork,
+} as const satisfies AnvilChain;
+
+//  PORT: 8546
+const baseAnvil = {
+  ...base,
+  ...getRpcUrls({ port: 8546 }),
+  fork: baseFork,
+} as const satisfies AnvilChain;
+
+//  PORT: 8547
+const mainnetAnvil = {
+  ...mainnet,
+  ...getRpcUrls({ port: 8547 }),
+  fork: mainnetFork,
+} as const satisfies AnvilChain;
+
+//  PORT: 8548
+const arbitrumAnvil = {
+  ...arbitrum,
+  ...getRpcUrls({ port: 8548 }),
+  fork: arbitrumFork,
+} as const satisfies AnvilChain;
+
+const account = privateKeyToAccount(PRIVATE_KEY);
+
+// TEST (CHAIN) CLIENTS
+
+export const chainClientMainnet: ChainClient = createTestClient({
+  account,
+  chain: mainnetAnvil,
+  mode: "anvil",
+  transport: http(),
+})
+  .extend(forkMethods)
+  .extend(publicActions)
+  .extend(walletActions);
+
+export const chainClientArbitrum: ChainClient = createTestClient({
+  account,
+  chain: arbitrumAnvil,
+  mode: "anvil",
+  transport: http(),
+})
+  .extend(forkMethods)
+  .extend(publicActions)
+  .extend(walletActions);
+
+// export const chainClientOptimism: OpStackChainClient = createTestClient({
+//   account,
+//   chain: optimismAnvil,
+//   mode: "anvil",
+//   transport: http(),
+// })
+//   .extend(forkMethods)
+//   .extend(publicActionsL1())
+//   .extend(publicActionsL2());
+
+// export const chainClientBase: OpStackChainClient = createTestClient({
+//   account,
+//   chain: baseAnvil,
+//   mode: "anvil",
+//   transport: http(),
+// })
+//   .extend(forkMethods)
+//   .extend(publicActionsL1())
+//   .extend(publicActionsL2());
+
+// PUBLIC CLIENTS
+// export const publicClientOptimism = createPublicClient({
+//   chain: optimismAnvil,
+//   transport: http(),
+// });
+
+// export const publicClientBase = createPublicClient({
+//   chain: baseAnvil,
+//   transport: http(),
+// });
```

Comment:
Why are these commented?

Reply:
test clients with op stack chains behave quite differently and so the computed type ChainClient cannot reliably be shared. I will just delete these instead since I don't think we need these.